contracts/build.go

```diff
@@ -0,0 +1,127 @@
+package contracts
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/integrations-framework/tools"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	DefaultBindPkg = "ethereum"
+)
+
+// ContractBuilder is an interface to build different chains contracts
+type ContractBuilder interface {
+	Targets() ([]*ContractTarget, error)
+	Compile(targets []*ContractTarget) error
+	GenerateBindings(targets []*ContractTarget) error
+}
+
+// ContractTarget contract build target
+type ContractTarget struct {
+	Version   string
+	Contracts []string
+}
+
+// abigenArgs is the arguments to the abigen executable. E.g., Bin is the -bin
+// arg.
+type abigenArgs struct {
+	ExecutablePath, Bin, ABI, Out, Type, Pkg string
+}
+
+// EthereumContractBuilder is a builder struct for Ethereum
+type EthereumContractBuilder struct {
+	ExecutablePath string
+}
+
+// NewEthereumContractBuilder builds solidity contracts
+func NewEthereumContractBuilder(execPath string) *EthereumContractBuilder {
+	return &EthereumContractBuilder{ExecutablePath: execPath}
+}
+
+// Compile compiles contracts
+func (b *EthereumContractBuilder) Compile(_ []*ContractTarget) error {
+	// step is not needed for Ethereum, already has ABI/BIN build in CI
+	return nil
```

Comment:
We already have contracts compilation there https://github.com/smartcontractkit/chainlink/blob/develop/.github/workflows/continuous-integration-workflow.yml#L67 can skip this step and generate bindings from ABI/BIN only.